src/app/Processor/ProcessorDetailPage/ProcessorDetailPage.tsx

```diff
@@ -83,21 +112,87 @@ const ProcessorDetailPage = (): JSX.Element => {
       console.error(processorError);
       goToInstance();
     }
-  }, [bridgeError, processorError, goToHome, goToInstance]);
+    if (updateProcessorError) {
+      console.error(updateProcessorError);
+    }
+  }, [
+    bridgeError,
+    processorError,
+    goToHome,
+    goToInstance,
+    updateProcessorError,
+  ]);
+
+  const processorNotChanged = useCallback(
+    (prevDef: ProcessorResponse, updatedDef: ProcessorRequest): boolean =>
+      isEqualWith(
+        prevDef,
+        updatedDef,
+        (prev: ProcessorResponse, next: ProcessorRequest) =>
+          isEqual(prev.name, next.name) &&
+          isEqual(prev.filters, next.filters) &&
+          isEqual(prev.transformationTemplate, next.transformationTemplate) &&
+          isEqual(prev.action, next.action) &&
+          isEqual(prev.source, next.source)
+      ),
+    []
+  );
+
+  const handleUpdateProcessorSaving = useCallback(
+    (processorRequest: ProcessorRequest) => {
+      if (
+        currentProcessor &&
+        processorNotChanged(currentProcessor, processorRequest)
+      ) {
+        setIsEditing(false);
+        return;
+      }
+      void updateProcessor(instanceId, processorId, processorRequest).then(() =>
+        setIsEditing(false)
+      );
```

Comment:
Minor thing. I guess you could do `setIsEditing(false)` inside the `useEffect` that handles `updatedProcessor` (lines 102~104)

Reply:
Ok, then we can get rid of returning a Promise